src/core/core.scale.js

```diff
@@ -668,16 +676,11 @@ module.exports = function(Chart) {
 						context.lineTo(itemToDraw.tx2, itemToDraw.ty2);
 					}
 
-					if (gridLines.drawOnChartArea) {
-						context.moveTo(itemToDraw.x1, itemToDraw.y1);
-						context.lineTo(itemToDraw.x2, itemToDraw.y2);
-					}
-
 					context.stroke();
 					context.restore();
 				}
 
-				if (optionTicks.display) {
```

Comment:
i understand, i'll add just add the statement to skip the drawing of label itself if it's empty string then